thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/scheduler/DetectionCronScheduler.java

```diff
@@ -79,28 +81,45 @@ public void run() {
 
       // add or update
       for (DetectionConfigDTO config : configs) {
-        JobKey key = new JobKey(getJobKey(config.getId()), TaskConstants.TaskType.DETECTION.toString());
         if (!config.isActive()) {
-          LOG.debug("Detection config " + key + " is inactive. Skipping.");
+          LOG.debug("Detection config " + config.getId() + " is inactive. Skipping.");
           continue;
         }
-        if (DetectionUtils.isDataAvailabilityCheckEnabled(config)) {
-          LOG.debug("Detection config " + key + " is enabled for data availability scheduling. Skipping.");
+        if (config.isDataAvailabilitySchedule()) {
+          LOG.debug("Detection config " + config.getId() + " is enabled for data availability scheduling. Skipping.");
           continue;
         }
+
         try {
-          if (scheduler.checkExists(key)) {
-            LOG.info("Detection config  " + key.getName() + " is already scheduled");
-            if (isJobUpdated(config, key)) {
-              // restart job
-              stopJob(key);
-              startJob(config, key);
+          // Schedule detection jobs
+          JobKey detectionJobKey = new JobKey(getJobKey(config.getId(), TaskConstants.TaskType.DETECTION),
+              QUARTZ_DETECTION_GROUPER);
+          JobDetail detectionJob = JobBuilder.newJob(DetectionPipelineJob.class).withIdentity(detectionJobKey).build();
+          if (scheduler.checkExists(detectionJobKey)) {
+            LOG.info("Detection config " + detectionJobKey.getName() + " is already scheduled for detection");
+            if (isJobUpdated(config, detectionJobKey)) {
+              restartJob(config, detectionJob);
+            }
+          } else {
+            startJob(config, detectionJob);
+          }
+
+          // Schedule data quality jobs
+          JobKey dataQualityJobKey = new JobKey(getJobKey(config.getId(), TaskConstants.TaskType.DATA_QUALITY),
+              QUARTZ_DETECTION_GROUPER);
+          JobDetail dataQualityJob = JobBuilder.newJob(DataQualityPipelineJob.class).withIdentity(dataQualityJobKey).build();
```

Comment:
will there be a case that the detection and data quality check has different corn?

Reply:
By design the detection config supports only 1 cron. If we need a different cron, then we can setup another alert.